### DISCUSSION TITLE
G7 state isn't currently known. Next connection will update this

### DISCUSSION THREAD

**Discussion by daplew** [NightscoutFoundation/xDrip#3771] [Q&A]
Hi, 
Wondering if anyone could help. I had a G7 working fine for 3 days after insertion and then overnight on day 3-4 it lost the connection. It gives the error: "G7 state isn't currently known. Next connection will update this". There hasn't been a connection for 12 hours now. 

All settings have been verified to match with guidance https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

Software version is on the stable channel: 2124091707

System status page says Brain State: Initializing Errors, Phone Service State: Authorizing

Any thoughts anyone?

**Comment by EasternPA** [comment]
Wife went through this for the second time two days ago. She came to me when it was missing readings for over 400 minutes. Only fix was to delete it from Bluetooth and start over in xDrip+. I deleted the sensor code, switched the data collector to none, then back to Dexcom and reentered the code in the field. I probably rebooted the phone for good measure as well. It's been fine ever since.

**Reply by daplew** [reply]
Thanks, I'll give this a go shortly!

**Reply by Navid200** [reply]
And you will be wasting your time if battery settings are incorrect.

**Reply by KWReid** [reply]
> Wife went through this for the second time two days ago. She came to me when it was missing readings for over 400 minutes. Only fix was to delete it from Bluetooth and start over in xDrip+. I deleted the sensor code, switched the data collector to none, then back to Dexcom and reentered the code in the field. I probably rebooted the phone for good measure as well. It's been fine ever since.

What you'd described did work for me.  I didn't need to restart phone.
1. Remove BT connection to the sensor on phone
2. Changed 4 digit sensor ID in Xdrip (just for funzies) 
3. Changed data collector to NONE
4. Changed data back to DEX
5. Re-entered correct sensor code
6. Restarted collector on xDrip.
... waited...

and it triggered a new pairing request and connected and resumed operation successfully.

**Comment by Navid200** [comment]
What is the phone brand and model?
Post a screenshot of the Dex Status page.

**Reply by daplew** [reply]
![Screenshot_20241112-173505.png](https://github.com/user-attachments/assets/323d4e4b-75c4-4df9-a09a-d2141ad09087)

This is on a Google Pixel 7A running Android 14 (AP2A.240905.003)

**Reply by Navid200** [reply]
Verify that you have disabled xDrip battery optimization:
https://navid200.github.io/xDrip/docs/BatteryOpt.html 

Do the same for Bluetooth: 
https://navid200.github.io/xDrip/docs/BluetoothBatteryOpt.html 

In both cases, make sure you se unrestricted checked.

**Reply by daplew** [reply]
![Screenshot_20241112-175419.png](https://github.com/user-attachments/assets/4e9d4cfc-d2d9-4ef5-9efb-e2ebc3c0defb)

![Screenshot_20241112-182243.png](https://github.com/user-attachments/assets/3354dd81-438c-41c8-8c01-609c2f76c208)

![Screenshot_20241112-182259.png](https://github.com/user-attachments/assets/bf65f153-7ebf-419c-9016-bccefaa473ee)

All verified

**Reply by Navid200** [reply]
You need to tap on allow background usage so that you access the page.

**Reply by daplew** [reply]
Thanks. All unrestricted except Bluetooth Legacy which was optimised - is this the culprit?

**Reply by Navid200** [reply]
Yes, it is.
Now, assuming all settings are correct, meaning aggressive service restart and all Bluetooth watchdogs, all you need to do is to restart the phone.

Let me know if it still does not reconnect.

**Reply by uziaharon16** [reply]
I'm facing similar issue with Pixel 7a, Dexcom G7, XDrip Version of Nov. 26, 2024.
Battery settings for XDrip+, Bluetooth and Bluetooth Legacy are set to Unrestricted.
Already gone through settings multiple times.
Already enabled "Special Pairing Mode".
Enabled today "Avoid Scanning".

The screenshots are before enabling "Avoid Scanning".

Any additional recommendation?

![IMG-20241231-WA0001](https://github.com/user-attachments/assets/8a756949-55d9-4b15-be35-67ddbeea45da)

![IMG-20241231-WA0002](https://github.com/user-attachments/assets/237bfc81-f705-4760-8574-b131d5e0c471)

**Reply by Navid200** [reply]
Sensor errors are not connectivity errors.  They are caused by the sensor portion.  
If you temporarily switch to Dexcom app, you will have the same outcome, no readings.
It could resolve itself in a few hours or not.

If you have a connectivity issue caused by xDrip, switching to Dexcom app will provide readings in Dexcom app.

**Reply by uziaharon16** [reply]
Thanks for your quick response!
So basically there are no settings which can overcome this?
Should I disable "Special Paring Workaround" and "Avoid Scanning" ?

**Reply by Navid200** [reply]
I would disable avoid scanning and special pairing workaround for a Pixel.

I once got sensor errors while I was snowshoeing.  When I got back in the car, the readings returned.  The gap was never backfilled, which was another evidence that this was not a connectivity issue.

The problem with sensor errors is that there is no way to predict if readings will come back or not.

**Reply by Navid200** [reply]
Was there a sudden change of temperature?
Was there a period of compression?
Was there an accidental hit to the sensor from a door frame and anything else?

**Reply by uziaharon16** [reply]
Well, it is a bit cold outside, but not that cold.
I don't think there was any compression, actually at night everything worked fine.
Also, no accidental hit to the sensor.
It's already the 4th sensor my son is facing issues with. 
We just started with G7, and didn't face such issues with G6, at least not at this frequency.
The first 2 sensors failed after 3 days.
The 3rd one was with errors for multiple days but somehow we made it to the 7th day.
Yesterday we connected the 4th one, I was so happy to find this discussion, since BT Legacy was not unrestricted.
I set it to unrestricted, and it worked great for 26 hours, but today it started with showing the errors.
It's still running now, but we have errors from time to time. 
It does not backfill the data as you mentioned above.
My son is thin and does not have a lot of fat tissue. He's applying the sensor to the thigh (same as he did with G6) thought it's not recommended, but he's not willing to change the location :(

**Reply by Navid200** [reply]
Some claim staying hydrated may help.
I have always worn my sensors on my arms, G6 and G7.

This is not directly related.  But, just to let you know, you can use Dexcom app instead of xDrip and have xDrip on the same phone get readings from Dexcom app.
I only use xDrip because I prefer to have only one app.  But, just to let you know.

There may be groups/forums that talk about the differences and similarities of G6 and G7 and common G7 problems etc. that may be helpful.

**Reply by uziaharon16** [reply]
Thank you so much for your help and all your work!
Happy new year!

**Reply by Mocadon** [reply]
Just an FYI, I never had a G6 sensor fail over several years. Then, in 2024, I had 3 G7 sensors fail over a 2 month period. I was running the Dexcom G7 byod app but after doing some research and talking to Dexcom support it was clearly some bad batches of sensors. Dexcom didn't have a problem sending me free replacements. Since that episode everything's been fine.

**Comment by horacioofman** [comment]
Hi. Sorry if writing on this thread is wrong. I am having the same issue. "G7stat e isn't currently known". I have deleted all the Bluetooth connections and review most of the settings. The sensor was working and after 3 days something failed and I could not connect sinvñce then. I have replaced my sensor, but now xdrip is not even asking me to pair the new sensor. My phone is a pixel 8

.![Screenshot_20251002-112516.png](https://github.com/user-attachments/assets/5f768e2f-6ea8-4281-8d6b-3c7d93ef2301)

![Screenshot_20251002-112531.png](https://github.![Screenshot_20251002-112541.png](https://github.com/user-attachments/assets/8e79d2ab-122a-48bf-9a40-57aff534a38c)

0b-50ea1a767c37)

![Uploading Screenshot_20251002-112541.png …]()

**Reply by horacioofman** [reply]
![Screenshot_20251002-112541.png](https://github.com/user-attachments/assets/8781bf96-2fba-4df4-8cad-cfe45f416d78)

**Reply by horacioofman** [reply]
the only setting I am not able to check is the bluetooth battery usage. Whenever i look for the system apps, bluetooth is not there. It is worth noting that I have been using G7 and Stelo on this phone for a long while and never had any conectivity issues. So bluetooth settings were probably fine.
I cannot figure out what is the problem. I would apreciate any help. I think I have tried all the reccomendations in this thread.
Thanks

**Reply by horacioofman** [reply]
![Screenshot_20251002-114923.png](https://github.com/user-attachments/assets/6a3d8ec4-2fd2-437b-9f9b-7144729b7d8d)

**Reply by horacioofman** [reply]
![Screenshot_20251002-114928.png](https://github.com/user-attachments/assets/fc53522a-4fac-4b5a-be03-034a095b79a2)

**Reply by horacioofman** [reply]
The sensor is a Stelo

**Comment by Navid200** [comment]
@horacioofman It's much better if you open a new thread.  
There are so many different factors that could lead to similar symptoms.  There is no reason to use the same thread for all.
It will only make it harder for someone who has problems to go through convoluted threads trying to figure out what they need to do.
So, I provide a link to the troubleshooting page.  But, if you need more help after looking at it, please don't post here.  Please open a new thread and I will be more than happy to assist you if I can.

https://navid200.github.io/xDrip/docs/Dexcom/G7_Troubleshooting.html

**Reply by horacioofman** [reply]
Thanks man!. Will try your troubleshooting link, and open a new thread if I still have the issue.
